### DISCUSSION TITLE
Pass culture along in the IDynamicRootQueryStep

### DISCUSSION THREAD

**Discussion by jemayn** [umbraco/Umbraco-CMS#15097] [Features and ideas]
Been playing around a bit with creating custom IDynamicRootQueryStep, and I was thinking it may be nice to know the current culture that the user is working from.

May be nice to fx from an archive to be able to target different sites depending on the culture of the current node.

**Comment by bergmania** [comment]
I like the idea, but it would require the frontend to do the request for each variant in the splitview. 

Furthermore, I would prefer it is was just available by injecting the `IVariationContextAccessor` instead of as injected parameters

**Reply by jemayn** [reply]
If injecting IVariationContextAccessor worked that would definitely fit my original usecase!

**Reply by bergmania** [reply]
We have the information available in the header, expect when using split view. Personally I can't figure out how to change the frontend code to check this for each culture in the split view.

Can you elaborate on your use case? I try to figure out how common it is?

**Reply by jemayn** [reply]
Sure! In this case we have a site with products in 3 languages. The problem is that the 3 language websites are wildy different, so we have 3 seperate root nodes, 1 for each language:

```
- Site lang1
-- Homepage
-- Categories
--- Category
- Site lang2
- Site lang3
- Archive
-- Products
--- Product
```

So sites and all their subpages are not variant, but stuff in the archive like the products are.

On a product you can pick a category, and based on that relation we get all products under a category and render on the page automatically.

So the category -> product variant is handled directly on the product since that is where all other product info is managed and allows the editors to stay within context.

Currently whenever setting the relevant categories the editor needs to navigate from the site root to the right site and select the relevant categories.

With the new QueryStep and a variant context we could make it so based on the products culture we could automatically map to the categories list on the corresponding site.

Hope that made sense 😅

**Reply by bergmania** [reply]
This PR https://github.com/umbraco/Umbraco-CMS/pull/15168 will handle the server side. But it will still not work on split view.
